api.md

```diff
@@ -170,6 +171,114 @@ SELECT add_dimension('conditions', 'device_id', number_partitions => 2);
 SELECT add_dimension('conditions', 'device_id', number_partitions => 2, if_not_exists => true);
 ```
 
+---
+## add_data_node() [](add_data_node)
```

Comment:
For this type of distributed hypertable / distributed TimescaleDB management, I think we need to create some canonical instructions about where they are executing, and possibly even capture this with the arguments.  For example, I *think* this is executed on the access node which is handling DDL for the distributed hypertable, but this should be clarified?

Also, I find it gets a little confusing between "database" and "bootstrap_database".  Both seem to refer to the name on the remote data node where the the distributed hypertable will be instantiated.

Similarly, to nit pick, are we _creating_ a "remote" hypertable, or is this about expanding an existing distributed hypertable?

Finally, I don't fully understand IP address / hostname management from this function, given that hostname appears to be _optional_ from these instructions.  What happens if the hostname isn't provided?  Minor q:  Is there support for submitting IP addresses vs. hostnames (or is it just passing in IPv4/IPv6 as a string to host argument?)

Reply:
Now reading the same usage below, seems like this is meant to be used _below_ we actually create the distributed hypertable.  I think this could be better clarified in the preface to this API call?

Similarly, please clarify is this is _also_ used for elastically expanding an existing hypertable (i.e., you can in the middle of existing hypertable, add_data_node() then expand hypertable to that data node), or if that's a different API call.

Reply:
This is executed on the access node and yes, there are no general instructions on how distributed hypertables work, which would help in making the role of this particular function clear.

I agree that all the users, passwords, and hosts is a little confusing. This seems to be an intermediate situation and will be resolved as we go forward. IMHO, we should still document it in it's current state though.

The function `attach_data_node()` is used to elastically scale an existing hypertable, as clarified by @erimatnor below. Followed his suggestion and updated the text.

Reply:
@mkindahl Note PR here: https://github.com/timescale/timescaledb-private/pull/193

It refactors and simplifies this function. Also, it generally hardens the server management functions with proper permissions checks. 

You need USAGE on servers to add servers and owner on a hypertable to be able to attach servers to it. Might also be worth mentioning.

Reply:
@erimatnor Yeah, but I think we can merge this PR in the current state and fix things as we go, adding a documentation task after the PR above is merged.